packages/eas-cli/src/commands/share.ts

```diff
@@ -0,0 +1,305 @@
+): Promise<string> {
+  const applicationArchivePatternOrPath =
+    inputBuildPath ??
+    (platform === Platform.ANDROID
+      ? 'android/app/build/outputs/**/*.{apk,aab}'
+      : 'ios/build/Build/Products/*simulator/*.app');
+
+  let applicationArchives = await findArtifactsAsync({
+    rootDir: process.cwd(),
+    patternOrPath: applicationArchivePatternOrPath,
+  });
+
+  if (applicationArchives.length === 0 && !inputBuildPath) {
+    Log.warn(`No application archives found at ${applicationArchivePatternOrPath}.`);
+    const { path } = await promptAsync({
+      type: 'text',
+      name: 'path',
+      message: 'Provide a path to the application archive:',
+      validate: value => (value ? true : 'Path may not be empty.'),
+    });
+    applicationArchives = await findArtifactsAsync({
+      rootDir: process.cwd(),
+      patternOrPath: path,
+    });
+  }
+
+  if (applicationArchives.length === 1) {
+    return applicationArchives[0];
+  }
+
+  if (applicationArchives.length > 1) {
+    const { path } = await promptAsync({
+      type: 'select',
+      name: 'path',
+      message: 'Found multiple application archives. Select one:',
+      choices: applicationArchives.map(archivePath => {
+        return {
+          title: archivePath,
+          value: archivePath,
+        };
+      }),
+    });
+    return path;
+  }
+
+  throw new Error(`Found no application archives at ${inputBuildPath}.`);
+}
+
+async function findArtifactsAsync({
+  rootDir,
+  patternOrPath,
+}: {
+  rootDir: string;
+  patternOrPath: string;
+}): Promise<string[]> {
+  const files: string[] = path.isAbsolute(patternOrPath)
+    ? (await fs.pathExists(patternOrPath))
+      ? [patternOrPath]
+      : []
+    : await fg(patternOrPath, { cwd: rootDir, onlyFiles: false });
+
+  return files.map(filePath => {
+    // User may provide an absolute path as input in which case
+    // fg will return an absolute path.
+    if (path.isAbsolute(filePath)) {
+      return filePath;
+    }
+
+    // User may also provide a relative path in which case
+    // fg will return a path relative to rootDir.
+    return path.join(rootDir, filePath);
+  });
+}
+
+async function uploadAppArchiveAsync(
+  graphqlClient: ExpoGraphqlClient,
+  path: string
+): Promise<string> {
+  const fileSize = (await fs.stat(path)).size;
+  const bucketKey = await uploadFileAtPathToGCSAsync(
+    graphqlClient,
+    UploadSessionType.EasShareGcsAppArchive,
+    path,
+    createProgressTracker({
+      total: fileSize,
+      message: 'Uploading to EAS Share',
+      completedMessage: 'Uploaded to EAS Share',
+    })
+  );
+  return bucketKey;
+}
+
+type AppMetadata = {
+  fingerprintHash?: string;
+  developmentClient: boolean;
+  simulator: boolean;
+};
+
+async function extractAppMetadataAsync(
+  buildPath: string,
+  platform: Platform
+): Promise<AppMetadata> {
+  let developmentClient = false;
+  let fingerprintHash: string | undefined;
+  const simulator = platform === Platform.IOS;
```

Comment:
is it even a valid use case to share `.ipa` builds? I'm just curious

Reply:
That's right, let's not support .ipa's for now